src/graphql/candyMachine.ts

```diff
@@ -15,24 +15,223 @@ import fs from 'fs/promises';
+  const walletKeyPair = web3.Keypair.fromSecretKey(Uint8Array.from(bytes));
+  const connection = new Connection(rpc);
+  const metaplex = new Metaplex(connection);
+  metaplex.use(keypairIdentity(walletKeyPair));
+  if (env === 'localnet') {
+    metaplex.use(ammanMockStorage('amman-mock-storage'));
+  } else {
+    if (config['storage'] !== 'nft-storage' || !config['nftStorageKey']) {
+      const message =
+        'hidden settings currently only works with nft storage and requires an nftStorageKey';
+      logger.error(message);
+      throw Error(message);
+    }
+
+    metaplex.use(
+      nftStorage({
+        token: config['nftStorageKey'],
+      }),
+    );
+  }
+
+  const storageDriver = metaplex.storage();
+
+  // First we need to upload the necessary NFT files from the zip
+  const zipFilesDir = await downloadZip(filesZipUrl, processId);
+  const templateNftPath = path.join(zipFilesDir, '0.json');
+  const templateNftMetadataStr = await fs.readFile(templateNftPath, 'utf-8');
+  let templateNftMetadata = JSON.parse(templateNftMetadataStr);
+  logger.info(templateNftMetadata);
+
+  // files look like:
+  // { uri: "0.png", type: "image/png" },
+  // { uri: "0.mp4", type: "video/mp4" },
+  const files = templateNftMetadata['properties']['files'];
+
+  for (let nftFile of files) {
+    const fileName = nftFile['uri'];
+    const filePath = path.join(zipFilesDir, fileName);
+    const fileBytes = await fs.readFile(filePath);
+    const contentType = contentTypeForFileName(fileName);
+    const metaplexFile = toMetaplexFile(fileBytes, fileName, {
+      contentType: contentType,
+      extension: fileName.split('.')[-1],
+    });
+    const uploadedFileUri = await storageDriver.upload(metaplexFile);
+
+    // update the nft metadata with uploaded uri
+    nftFile['uri'] = uploadedFileUri;
+    if (templateNftMetadata['image'] === fileName) {
+      templateNftMetadata['image'] = uploadedFileUri;
+    } else if (templateNftMetadata['animation_url'] == fileName) {
+      templateNftMetadata['animation_url'] = uploadedFileUri;
+    }
+  }
+
+  logger.info('Uploading NFT Metadata: \n', templateNftMetadata);
+  const nftMetadataUploadedFileUri = await storageDriver.uploadJson(
+    templateNftMetadata,
+  );
+
+  logger.info(nftMetadataUploadedFileUri);
+
+  config['hiddenSettings'] = {
+    name: 'My Nft', // TODO
+    uri: nftMetadataUploadedFileUri,
+    // all zeroes should be fine since all the NFT's are the same
+    // in cases where nfts have different traits we will want to update this
+    // to be the hash of a file that specifies the mint order
+    hash: new Array(32).fill(0),
+  };
+
+  const pubkeyFields = ['solTreasuryAccount', 'collection'];
+  for (const field of pubkeyFields) {
+    if (!!config[field]) {
+      config[field] = new PublicKey(config[field]);
+    }
+  }
+
+  config['authority'] = walletKeyPair;
+  config['price'] = sol(config['price'] as number);
+
+  // now the config is ready
```

Comment:
is there a better comment you can put here? This one doesnt help me understand the code

Reply:
done